Enhance sfx_handler to accomodate more formats

ChinYikMing opened the pull request:
Quake's WAV format is different from Doom's WAV format, so we must improve the sfx_handler by determinating the format type before playing sound effect in order to accommodate Quake's WAV format.

jserv commented on docs/syscall.md:
```diff
@@ -174,7 +174,7 @@ The request is similar to how music is managed earlier in the description and su
 
 #### Sound Effect(sfx)
-`sfxinfo_t` is a structure that defines sound effect data and size. The `data` and `size` fields of the structure are used to play sound effect with the SDL2_mixer library.
+`sfxinfo_t` is a structure that defines sound effect data and size. The `data` and `size` fields of the structure are used to play sound effect with the SDL2_mixer library. Currently, support sound effect of Doom's WAV format and normal WAV format(includes RIFF header).
```
Add a hyperlink to WAV and RIFF. Wikipedia is preferable.

jserv commented on src/syscall_sdl.c:
```diff
@@ -594,8 +598,10 @@ static void *sfx_handler(void *arg)
+    if (*ptr & 0x3) /* Doom, multiplied by 8 because sfx->volume's max is 15 */
+        Mix_Volume(chan, sfx->volume * 8);
+    else /* Quake, mod by 127 because sfx->volume's max is 255 */
+        Mix_Volume(chan, sfx->volume % 127);
```
Is it correct to perform `sfx->volume % 127`?

ChinYikMing replied:
Fixed it. It should be `(sfx->volume + 1) % 128` because `sfx->volume`'s max is 255 and `Mix_Volume`'s max is 128.

ChinYikMing committed:
Enhance sfx_handler to accomodate more formats

Quake's WAV format is different from Doom's WAV format,
so we must improve the sfx_handler by determinating the format type
before playing sound effect in order to accommodate Quake's WAV format.

jserv commented:
Thank @ChinYikMing for contributing!
